tests/exchange/test_exchange.py

```diff
@@ -3601,3 +3701,165 @@ def test__calculate_funding_fees_datetime_called(
     time_machine.move_to("2021-09-01 08:00:00 +00:00")
     funding_fees = exchange._calculate_funding_fees('ADA/USDT', 30.0, d1)
     assert funding_fees == expected_fees
+
+
+@pytest.mark.parametrize('pair,expected_size,trading_mode', [
+    ('XLTCUSDT', 1, 'spot'),
+    ('LTC/USD', 1, 'futures'),
+    ('XLTCUSDT', 0.01, 'futures'),
+    ('LTC/ETH', 1, 'futures'),
+    ('ETH/USDT:USDT', 10, 'futures')
+])
+def test__get_contract_size(mocker, default_conf, markets, pair, expected_size, trading_mode):
+    api_mock = MagicMock()
+    default_conf['trading_mode'] = trading_mode
+    default_conf['collateral'] = 'isolated'
+    mocker.patch('freqtrade.exchange.Exchange.markets', markets)
+    exchange = get_patched_exchange(mocker, default_conf, api_mock)
+    size = exchange._get_contract_size(pair)
+    assert expected_size == size
```

Comment:
the thing i don't like about this test is that it takes assumptions about the setup in conftest - and basically only tests that - and only partially the logic it should be testing.

while we do this in other areas too, this is explicitly testing for the values contained there, so i don't think a better approach here would be: 
instead of pair - provide contractSize.
Set that for a (static) pair as part of the test in a dummy market structure - and then verify if the result is expected - basically testing if "nothing" or spot (or 1)  returns 1 - and the other cases return the provided value.

That'll remove the dependency on the "markets" - which can fall on our feet should we ever decide to use these pairs for some other tests (it did create problems in the past already - so i think new tests should not create new dependencies in this regards).

Reply:
I don't understand what you meant in your comment, but the test no longer assumes values from conftest